### PR TITLE
bug(mapper/product_mapper): update inventory tracking condition

### PR DESCRIPTION
**Pull request by Jhalter5Stones**
update inventory tracking condition

### CODE REVIEW

**Comment by Jhalter5Stones** [comment]
@mrogers-5s Ready for review

https://trello.com/c/kmkF8JKj/540-usbat-see-correct-inventory-tracking-setting-for-all-products

File: lib/mapper/product_mapper.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def self.map_payload(product, additional_data = {}, track_inventory = true, defa
                   type: isDigital ? 'digital' : 'physical',
                   weight: product['weight'] ? product['weight']['value'] : '0',
                   width: product['width'] ? product['width']['value'] : '0',
-                  inventory_tracking: isDigital || !track_inventory ? 'none' : 'product',
+                  inventory_tracking: isDigital || !track_inventory || product['nonInventory'] || product['assembly'] ? 'none' : 'product',
```
Comment (mrogers-5s) [inline lib/mapper/product_mapper.rb:33]: @Jhalter5Stones This agent should only have to check one field now. The acumen agent should set something like `trackInventory` or `shouldTrackInventory`, and this agent should look at that field.

Comment (Jhalter5Stones) [reply lib/mapper/product_mapper.rb]: @mrogers-5s Fixed.

File: lib/mapper/product_mapper.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@ def self.map_payload(product, additional_data = {}, track_inventory = true, defa
                   type: isDigital ? 'digital' : 'physical',
                   weight: product['weight'] ? product['weight']['value'] : '0',
                   width: product['width'] ? product['width']['value'] : '0',
-                  inventory_tracking: isDigital || !track_inventory ? 'none' : 'product',
+                  inventory_tracking: isDigital || !track_inventory || !product['trackInventory'] ? 'none' : 'product',
```
Comment (mrogers-5s) [inline lib/mapper/product_mapper.rb:33]: @Jhalter5Stones You should use the boolify util here around `product['trackInventory']` Due to the agents in the flow, that value may come in as `"false"` which is truthy as it's a non-blank string

Comment (Jhalter5Stones) [reply lib/mapper/product_mapper.rb:33]: @mrogers-5s How would it come through as a string? I tested a payload through the flow and it came through as a boolean and not a string.

**Review by mrogers-5s — CHANGES_REQUESTED** [review]

File: lib/mapper/product_mapper.rb (hunk 2 above)
Comment (Jhalter5Stones) [reply lib/mapper/product_mapper.rb]: @mrogers-5s Ok can you take a look and see if this looks better?

Comment (Jhalter5Stones) [reply lib/mapper/product_mapper.rb]: @mrogers-5s I made updates to this. Can you take a look?

**Review by mrogers-5s — APPROVED** [review]

**Pull request merged**